lib/schema.js

```diff
@@ -77,23 +77,73 @@ const processSchema = (
   }
 };
 
+const getCategoryType = (category) => {
+  const type = category.constructor.name;
+  return type === 'Object' ? 'Local' : type;
+};
+
+const verifyLink = (
+  source,
+  sourceName,
+  destination,
+  destinationName,
+  propertyName
+) => {
+  const sourceType = getCategoryType(source);
+  const destinationType = getCategoryType(destination);
+
+  if (destinationType === 'Log') {
+    return `Illegal reference to Log from ${sourceName}.${propertyName}`;
+  }
+
+  if (destinationType === 'Local' && sourceType !== 'Local') {
+    return `Illegal reference to Local category ${destination} from ` +
+      `${propertyName} ${sourceName}.${propertyName}`;
+  }
+
+  return null;
```

Comment:
`return undefined;` or simply remove this line

Reply:
This function now returns an object, so `null` is better.